doi/src/main/java/ca/nrc/cadc/doi/PostAction.java

```diff
@@ -463,11 +474,78 @@ private void performDoiAction() throws Exception {
             log.debug("redirectUrl: " + redirectUrl);
             syncOutput.setHeader("Location", redirectUrl);
             syncOutput.setCode(303);
+        } else if (doiAction.equals(SEARCH_ACTION)) {
+            if (doiStatusSearchFilter == null || callersNumericId == null) {
+                getStatusList(getOwnedDOIList());
+            } else {
+                getStatusList(getFilteredDOIList(doiStatusSearchFilter));
+            }
         } else {
             throw new UnsupportedOperationException("DOI action not implemented: " + doiAction);
         }
     }
-    
+
+    private List<Node> getFilteredDOIList(DoiStatusSearchFilter doiStatusSearchFilter) throws Exception {
```

Comment:
Need to rethink this loop over the child nodes. If checking whether the node should be returned in the search results depends on the node, i.e. the node status or requestor ID, then that check needs to happen for each node.
But if the check doesn't depend on the node, i.e. is the user a member of the publisher group or is the user doiadmin, etc, that check should only happen once.